substratevm/src/com.oracle.svm.configure/src/com/oracle/svm/configure/config/SerializationConfiguration.java

```diff
@@ -85,18 +90,20 @@ public void registerWithTargetConstructorClass(ConfigurationCondition condition,
     }
 
     @Override
-    public void registerWithTargetConstructorClass(ConfigurationCondition condition, String className, String customTargetConstructorClassName) {
-        serializations.add(createConfigurationType(condition, className, customTargetConstructorClassName));
+    public void registerWithTargetConstructorClass(ConfigurationCondition condition, String rawClassName, String customTargetConstructorClassName) {
+        String className = rawClassName.contains(LambdaUtils.LAMBDA_FUNCTIONS_NAME_PATTERN) ? rawClassName.split(LambdaUtils.SPLIT_BY_LAMBDA)[0] + LambdaUtils.LAMBDA_FUNCTIONS_NAME_PATTERN
```

Comment:
I don't see a need for appending `$$Lambda$` here as you already have the boolean. I would keep the class name as is.

Reply:
I agree, fixed as suggested.